src/PHP.php

```diff
@@ -377,35 +377,136 @@ private function generate_default_conf( $cache_type, $server_name ) {
+			$launch = \EE::launch( sprintf( "docker exec %s bash -c \"ip route show default | cut -d' ' -f3\"", $container_name ) );
 
 			if ( ! $launch->return_code ) {
-				$this->site_data['db_host'] = trim( $launch->stdout, "\n" );
+				$db_host = trim( $launch->stdout, "\n" );
 			} else {
-				throw new \Exception( 'There was a problem inspecting network. Please check the logs' );
+				\EE::exec( "docker rm -f $container_name" );
+				throw new \Exception( 'There was a problem in connecting to the database. Please check the logs' );
 			}
 		}
 
 		\EE::log( 'Verifying connection to remote database' );
-		$img_versions = \EE\Utils\get_image_versions();
 
-		$network = ( GLOBAL_DB === $this->site_data['db_host'] ) ? "--network='" . GLOBAL_FRONTEND_NETWORK . "'" : '';
-
-		if ( ! \EE::exec( sprintf( "docker run -it --rm %s easyengine/mariadb:%s sh -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='EXIT'\"", $network, $img_versions['easyengine/mariadb'], $this->site_data['db_host'], $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'] ) ) ) {
+		$check_db_connection = sprintf(
+			"docker exec %s sh -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='EXIT'\"",
+			$container_name,
+			$db_host,
+			$this->site_data['db_port'],
+			$this->site_data['db_user'],
+			$this->site_data['db_password']
+		);
+		if ( ! \EE::exec( $check_db_connection ) ) {
+			\EE::exec( "docker rm -f $container_name" );
 			throw new \Exception( 'Unable to connect to remote db' );
 		}
-
 		\EE::success( 'Connection to remote db verified' );
+
+		$name            = str_replace( '_', '\_', $this->site_data['db_name'] );
+		$check_db_exists = sprintf( "docker exec %s bash -c \"mysqlshow --user='%s' --password='%s' --host='%s' --port='%s' '%s'\"", $container_name, $this->site_data['db_user'], $this->site_data['db_password'], $db_host, $this->site_data['db_port'], $name );
+
+		if ( ! \EE::exec( $check_db_exists ) ) {
+			\EE::log( sprintf( 'Database `%s` does not exist. Attempting to create it.', $this->site_data['db_name'] ) );
+			$create_db_command = sprintf(
+				"docker exec %s bash -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='CREATE DATABASE %s;'\"",
+				$container_name,
+				$db_host,
+				$this->site_data['db_port'],
+				$this->site_data['db_user'],
+				$this->site_data['db_password'],
+				$this->site_data['db_name']
+			);
+
+			if ( ! \EE::exec( $create_db_command ) ) {
+				\EE::exec( "docker rm -f $container_name" );
+				throw new \Exception( sprintf(
+					'Could not create database `%s` on `%s:%s`. Please check if %s has rights to create database or manually create a database and pass with `--dbname` parameter.',
+					$this->site_data['db_name'],
+					$this->site_data['db_host'],
+					$this->site_data['db_port'],
+					$this->site_data['db_user']
+				) );
+			}
+		} else {
+			if ( $this->force ) {
```

Comment:
Why this is not in `else if( $this->force) `? you can use `\EE::exec( "docker rm -f $container_name" );` in this else if and last else condition two time  ( execute only one time ) and avoid this big block of nesting.

I might have missed this in https://github.com/EasyEngine/site-type-wp/pull/73/files so we may need to do change in that.

Reply:
Because only two commands need to be run `if ( $this->force )` is true. Rest will come under the normal else condition. 
Also in the last else condition there is `\EE::exec( "docker rm -f $container_name" );` before throwing exception. So the below one will not execute if this is executed. Only one of them will execute in any case.